### PR TITLE
[IN-163][Emberosf] Add 'documentType' to 'preprint-provider' model.

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

Unify translation of 'preprintWord'. This PR is the basis for translation unification for both the Reviews and Preprints app.

## Summary of Changes

In `preprint-provider` model, the `i18n` service is injected. A new property `documentType` is added so that it gets all translation from the i18n service. Translations for preprint words then can be accessed like:
```
let preprintProvider = this.get('store').find('preprint-provider', 'osf');
preprint = this.get('preprintProvider.documentType.singular'); //'preprint'
Preprint = this.get('preprintProvider.documentType.singularCapitalized'); // 'Preprint'
preprints =this.get('preprintProvider.documentType.plural'); // 'preprints'
Preprints =this.get('preprintProvider.documentType.pluralCapitalized'); // 'Preprints' 
```

## Side Effects / Testing Notes

None.

## Ticket

https://openscience.atlassian.net/browse/IN-163

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: addon/locales/en/translations.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ export default {
         },
         navbar: {
             add: 'Add',
-            addAPreprint: 'Add a {{preprintWords.Preprint}}',
+            addAPreprint: 'Add a {{documentType}}',
```
Comment (aaxelb) [inline addon/locales/en/translations.js]: Drive-by comment:

It might be good to leave the translations file in charge of whether it wants capital/plural, and pass the whole `documentType` into `t`:
```hbs
{{t 'navbar.addAPreprint' documentType=preprint.provider.documentType}}
```
```js
addAPreprint: 'Add a {{documentType.singularCapitalized}}',
```
That way if someone decides we want a lower-case `p`, the change only happens in one place.

Comment (adlius) [reply addon/locales/en/translations.js]: Excellent idea. Will do that.
